### PR TITLE
boards/cc2538dk: use common Makefile.include

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The code in the board's `Makefile.include` is identical to [`common/remote/Makefile.include`](https://github.com/RIOT-OS/RIOT/blob/master/boards/common/remote/Makefile.include)
The scripts in [`dist/`](https://github.com/RIOT-OS/RIOT/tree/master/boards/common/remote/dist) are identical too.

Remove the duplicated code and use to the common Makefile/scripts instead.

### Testing procedure

You should still be able to flash the `cc2538dk`.
But textual comparison should be enough.

### CODE REVIEW

File: boards/cc2538dk/Makefile.include [hunk 1]
```diff
@@ -5,32 +5,8 @@ PROGRAMMER_SERIAL ?= 06EB
...
-OFLAGS = --gap-fill 0xff
-FLASHFILE ?= $(BINFILE)
-DEBUGGER_FLAGS = $(BINDIR) $(ELFFILE)
-RESET_FLAGS ?= $(BINDIR)
-
-OBJDUMPFLAGS += --disassemble --source --disassembler-options=force-thumb
-
-# include common cc2538 includes
-INCLUDES += -I$(RIOTBOARD)/common/cc2538/include
+include $(RIOTBOARD)/common/remote/Makefile.include
```
Comment (fjmolinas) [inline boards/cc2538dk/Makefile.include:12]: Can we move this to  `$(RIOTBOARD)/common/cc2538/Makefile.include` and have `openmote-b` use it as well, there is a couple of differences fro JLINK, but mostly they are the same and `PROG_BAUD ?= 460800` works as well for `remote-rev*`

Comment (fjmolinas) [reply boards/cc2538dk/Makefile.include:12]: I have all the affected boards so I can test afterwards.

**Comment by fjmolinas** [comment]
I tried this [patch](https://gist.github.com/fjmolinas/524abb300280ba860e9dcacdc26b3b5e) to put everything in common for `cc2538` boards, I tested on `cc2538dk` and `openmote-b`, the `remote-rev*` I have with me ATM does not have the jlink headers, I have another one at the office but wont be there until Monday, do you have one to test? Otherwise If this works for all paltforms IMO it will be better, I can also test on `openmote-cc2538` tomorrow:

- [x] `openmote-b`
- [x] `openmote-cc2538`
- [ ] `cc2538dk`
- [ ] `remote*`

**Comment by benpicco** [comment]
I do have a J-Link with the small header at home as well as an `openmote-b` - will test this evening.
Should I include the patch directly in this PR?

**Comment by fjmolinas** [comment]
> I do have a J-Link with the small header at home as well as an openmote-b - will test this evening.
> Should I include the patch directly in this PR?

If it works and you agree yes!

**Comment by benpicco** [comment]
I applied the patch, but now I get

```
% make -C examples/gnrc_networking BOARD=openmote-b   
make: Entering directory '/home/benpicco/dev/RIOT/examples/gnrc_networking'
/home/benpicco/dev/RIOT/examples/gnrc_networking/../../Makefile.include:527: *** FLASHFILE is not defined for this board: .  Stop.
make: Leaving directory '/home/benpicco/dev/RIOT/examples/gnrc_networking'
```

**Comment by fjmolinas** [comment]
> I applied the patch, but now I get

Seems like the renaming and new common dile did not apply in the patch, I updated the [gist](https://gist.github.com/fjmolinas/524abb300280ba860e9dcacdc26b3b5e)

**Comment by benpicco** [comment]
The first hunk doesn't apply.
Since this PR only touches `cc2538dk` we could also merge it as is and have you create a follow-up PR :wink:

**Comment by fjmolinas** [comment]
> The first hunk doesn't apply.
> Since this PR only touches cc2538dk we could also merge it as is and have you create a follow-up PR wink

The issue is that the common remote file also does: `INCLUDES += -I$(RIOTBOARD)/common/remote/include`, which is not necesarily wanted for `cc2538dk` right?

**Comment by benpicco** [comment]
superseded by #14724

**Pull request closed without merging**